### ISSUE TITLE
Datagrid columns should have the option to wrap cell contents like in Excel

### ISSUE THREAD

**Issue by prakhargupta1** [mui/mui-toolpad#1484]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Summary 💡

A table for me in preview mode looks like: 

<img width="1174" alt="Screenshot 2022-12-23 at 11 51 16 AM" src="https://user-images.githubusercontent.com/92228082/209282971-88979876-f8e4-41e4-a3b2-e35ac9f4ae69.png">

The only option to see the contents is by dragging the column header to the right and sometimes that also doesn't work when the text is too long.

**Comment by oliviertassinari** [comment]
I have created https://github.com/mui/mui-x/issues/7323 on MUI X's side.

**Comment by prakhargupta1** [comment]
There is a [dynamic row height](https://mui.com/x/react-data-grid/row-height/) feature that we can provide as config or have it by default.
In case of supporting it by default, we can explore getEstimatedRowHeight prop.

**Comment by Janpot** [comment]
> There is a [dynamic row height](https://mui.com/x/react-data-grid/row-height/) feature that we can provide as config or have it by default.
In case of supporting it by default, we can explore getEstimatedRowHeight prop.

Personally, I find the UX of datatables with varying row heights quite poor. It makes it hard to scan the data as your eyes have to travel a different distance for each row.
The `auto` feature is also an all-or-nothing approach. There is no way of constraining to a maximum row height, so one single large cell in a table can render it completely unusual. Or to have it operate on only a subset of the columns. It also makes resizing columns a bit prone to layout-shift in the rows.

I feel like we should rather look into https://github.com/mui/mui-x/issues/7323 to solve this problem

**Comment by oliviertassinari** [comment]
> The only option to see the contents is by dragging the column header to the right and sometimes that also doesn't work when the text is too long.

Shorterm, we could implement https://github.com/mui/mui-x/issues/1040#issuecomment-780484281

**Comment by bharatkashyap** [comment]
> > The only option to see the contents is by dragging the column header to the right and sometimes that also doesn't work when the text is too long.
> 
> Shorterm, we could implement a "1. Cell expander" [mui/mui-x#1040 (comment)](https://github.com/mui/mui-x/issues/1040#issuecomment-780484281)

Agreed, this is present in the docs as well now https://mui.com/x/react-data-grid/recipes-editing/#multiline-editing
